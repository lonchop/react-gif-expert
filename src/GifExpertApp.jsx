import { useState } from 'react'
import { AddCategory, GifGrid } from './components'
import { v4 as uuidv4 } from 'uuid';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch'])

  const onAddCategory = (newCategory) => {
    setCategories([newCategory, ...categories])
  }

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory onNewCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid
          key={uuidv4()}
          category={category}
        />
      ))}
    </>
  )
}
