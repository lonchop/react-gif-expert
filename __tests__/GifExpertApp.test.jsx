/* eslint-disable no-undef */
import { screen, render, fireEvent } from '@testing-library/react'
import { GifExpertApp } from '../src/GifExpertApp'
import { fetch } from 'whatwg-fetch'

describe('Pruebas en <GifExpertApp />', () => {
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifExpertApp />)
    screen.debug()
    expect(container).toMatchSnapshot()
  })
  test('Debe funcionar el input', () => {
    // Renderiza el componente
    // const { getByPlaceholderText } = render(<GifExpertApp />)
    render(<GifExpertApp />)


    // Obtiene el input por el placeholder
    // const input = getByPlaceholderText('Buscar gif')
    const input = screen.getByRole('textbox')

    // Simula escribir en el input
    fireEvent.change(input, { target: { value: 'Texto de prueba' } })

    // Verifica que el valor del input sea el esperado
    expect(input.value).toBe('Texto de prueba')
  })
})
