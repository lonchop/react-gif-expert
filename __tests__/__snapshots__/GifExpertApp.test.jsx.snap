// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gif"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
